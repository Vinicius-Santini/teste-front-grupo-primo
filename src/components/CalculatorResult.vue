<script setup>
import { selic, arca } from "../utils/yieldRates.js"

import CalculatorResultCard from "./CalculatorResultCard.vue"

const props = defineProps({
    investmentPeriod: Number,
    selicProfitability: Number,
    arcaProfitability: Number
})
</script>

<template>
    <div class="calculator-result-wrapper">
        <div class="calculator-result-container">
            <h1 class="calculator-result-title">
                Resultado:
            </h1>
            <h2 class="calculator-result-period">
                Em {{ investmentPeriod }} meses você teria:
            </h2>
            <CalculatorResultCard title="TAXA SELIC" :profitability=selicProfitability />
            <CalculatorResultCard logoPath="logo-arca.svg" title="FUNDO ARCA" :profitability=arcaProfitability />
            <div class="divider"></div>
            <div class="calculator-result-rates-wrapper">
                <div class="calculator-result-rate-content">
                    <div class="calculator-result-rate-name">TAXA SELIC:</div>
                    <div class="calculator-result-rate-value">{{ selic }}%</div>
                </div>
                <div class="calculator-result-rate-content">
                    <div class="calculator-result-rate-name">RENTABILIDADE DO ARCA:</div>
                    <div class="calculator-result-rate-value">{{ arca }}% a.a.</div>
                </div>
            </div>
            <div class="calculator-result-help-wrapper">
                <div class="calculator-result-help-content">
                    <img src="../assets/help.svg" class="" />
                    <div class="calculator-result-help-info">Valores utilizados no simulador de investimentos
                        (referentes à
                        data de última atualização - esses valores podem alterar de acordo com o mercado):</div>
                </div>
                <div class="calculator-result-help-date">- Data da última atualização: 10/01/2024</div>
            </div>
        </div>
    </div>
</template>

<style>
.calculator-result-wrapper {
    min-height: fit-content;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: var(--secondary-application-color);
    border-radius: 16px;
    color: var(--highlighted-text-color);
    font-weight: 600;

    @media (max-width: 1024px) {
        width: 90%;
    }

    .calculator-result-container {
        width: 94%;
        display: flex;
        flex-direction: column;
        align-items: flex-start;
        padding-top: 4vh;

        .calculator-result-title {
            font-weight: 700;
            font-size: 2.5rem;
            margin-bottom: 5vh;

            @media (max-width: 1024px) {
                font-size: 2rem;
            }
        }

        .calculator-result-period {
            font-size: 2rem;
            margin-bottom: 12px;
            color: #3A3A3A;

            @media (max-width: 1024px) {
                font-size: 1.125rem;
            }
        }

        .calculator-result-rates-wrapper {
            display: flex;
            flex-direction: column;
            align-items: flex-start;
            margin: 3vh 0 3vh 0;

            .calculator-result-rate-content {
                display: flex;
                align-items: center;
                margin-bottom: 3vh;

                @media (max-width: 1024px) {
                    flex-direction: column;
                    align-items: flex-start;
                }

                .calculator-result-rate-name {
                    font-size: 1.0625;
                    letter-spacing: 4px;
                    line-height: 19px;

                    @media (max-width: 1024px) {
                        font-size: 1rem;
                        line-height: 18px;
                    }
                }

                .calculator-result-rate-value {
                    font-weight: 800;
                    font-size: 1.5rem;
                    padding-left: 8px;
                    text-align: center;
                    line-height: 32px;

                    @media (max-width: 1024px) {
                        padding-top: 10px;
                        padding-left: 0px;
                        font-size: 1.125rem;
                        line-height: 24px;
                    }
                }
            }
        }

        .calculator-result-help-wrapper {
            font-weight: 400;
            font-size: 1.125rem;
            color: #6A6A6A;
            padding-bottom: 48px;
            display: flex;
            flex-direction: column;
            align-items: flex-start;
            text-align: justify;

            @media (max-width: 1024px) {
                font-size: 0.875rem;
            }

            .calculator-result-help-content {
                display: flex;
                align-items: center;

                @media (max-width: 1024px) {
                    align-items: flex-start;
                    margin-right: 20px;
                }

                img {
                    width: 32px;
                    height: 32px;
                    margin-right: 16px;
                }
            }

            .calculator-result-help-date {
                margin: 20px 0 0 40px;

                @media (max-width: 1024px) {
                    margin: 20px 40px 0 52px;
                }
            }
        }
    }

}
</style>