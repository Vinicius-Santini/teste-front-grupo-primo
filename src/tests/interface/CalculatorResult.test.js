import { test, expect } from "vitest";
import { render } from "@testing-library/vue";
import CalculatorResult from "../../components/CalculatorResult.vue";

test("CalculatorResult should render correctly", () => {
    const calculatorResult = render(CalculatorResult, { props: { investmentPeriod: 24, selicProfitability: 5000.0, arcaProfitability: 6000.0 } })

    expect(calculatorResult).toMatchSnapshot();
})