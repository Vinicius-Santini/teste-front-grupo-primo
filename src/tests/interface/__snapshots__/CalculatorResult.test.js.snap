// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CalculatorResult should render correctly 1`] = `
{
  "baseElement": <body>
    <div>
      <div
        class="calculator-result-wrapper"
      >
        <div
          class="calculator-result-container"
        >
          <h1
            class="calculator-result-title"
          >
             Resultado: 
          </h1>
          <h2
            class="calculator-result-period"
          >
             Em 24 meses você teria: 
          </h2>
          <div
            class="calculator-result-card-wrapper"
          >
            <div
              class="calculator-result-card-container"
            >
              <!--v-if-->
              <h3
                class="calculator-result-card-title"
              >
                TAXA SELIC
              </h3>
              <div
                class="calculator-result-card-value"
              >
                R$ 5.000,00
              </div>
            </div>
          </div>
          <div
            class="calculator-result-card-wrapper"
          >
            <div
              class="calculator-result-card-container"
            >
              <img
                aria-label="logo-arca.svg"
                class="calculator-result-card-logo"
                src="file:///C:/src/assets/logo-arca.svg"
              />
              <h3
                class="calculator-result-card-title"
              >
                FUNDO ARCA
              </h3>
              <div
                class="calculator-result-card-value"
              >
                R$ 6.000,00
              </div>
            </div>
          </div>
          <div
            class="divider"
          />
          <div
            class="calculator-result-rates-wrapper"
          >
            <div
              class="calculator-result-rate-content"
            >
              <div
                class="calculator-result-rate-name"
              >
                TAXA SELIC:
              </div>
              <div
                class="calculator-result-rate-value"
              >
                9.25%
              </div>
            </div>
            <div
              class="calculator-result-rate-content"
            >
              <div
                class="calculator-result-rate-name"
              >
                RENTABILIDADE DO ARCA:
              </div>
              <div
                class="calculator-result-rate-value"
              >
                18% a.a.
              </div>
            </div>
          </div>
          <div
            class="calculator-result-help-wrapper"
          >
            <div
              class="calculator-result-help-content"
            >
              <img
                class=""
                src="/src/assets/help.svg"
              />
              <div
                class="calculator-result-help-info"
              >
                Valores utilizados no simulador de investimentos (referentes à data de última atualização - esses valores podem alterar de acordo com o mercado):
              </div>
            </div>
            <div
              class="calculator-result-help-date"
            >
              - Data da última atualização: 10/01/2024
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calculator-result-wrapper"
    >
      <div
        class="calculator-result-container"
      >
        <h1
          class="calculator-result-title"
        >
           Resultado: 
        </h1>
        <h2
          class="calculator-result-period"
        >
           Em 24 meses você teria: 
        </h2>
        <div
          class="calculator-result-card-wrapper"
        >
          <div
            class="calculator-result-card-container"
          >
            <!--v-if-->
            <h3
              class="calculator-result-card-title"
            >
              TAXA SELIC
            </h3>
            <div
              class="calculator-result-card-value"
            >
              R$ 5.000,00
            </div>
          </div>
        </div>
        <div
          class="calculator-result-card-wrapper"
        >
          <div
            class="calculator-result-card-container"
          >
            <img
              aria-label="logo-arca.svg"
              class="calculator-result-card-logo"
              src="file:///C:/src/assets/logo-arca.svg"
            />
            <h3
              class="calculator-result-card-title"
            >
              FUNDO ARCA
            </h3>
            <div
              class="calculator-result-card-value"
            >
              R$ 6.000,00
            </div>
          </div>
        </div>
        <div
          class="divider"
        />
        <div
          class="calculator-result-rates-wrapper"
        >
          <div
            class="calculator-result-rate-content"
          >
            <div
              class="calculator-result-rate-name"
            >
              TAXA SELIC:
            </div>
            <div
              class="calculator-result-rate-value"
            >
              9.25%
            </div>
          </div>
          <div
            class="calculator-result-rate-content"
          >
            <div
              class="calculator-result-rate-name"
            >
              RENTABILIDADE DO ARCA:
            </div>
            <div
              class="calculator-result-rate-value"
            >
              18% a.a.
            </div>
          </div>
        </div>
        <div
          class="calculator-result-help-wrapper"
        >
          <div
            class="calculator-result-help-content"
          >
            <img
              class=""
              src="/src/assets/help.svg"
            />
            <div
              class="calculator-result-help-info"
            >
              Valores utilizados no simulador de investimentos (referentes à data de última atualização - esses valores podem alterar de acordo com o mercado):
            </div>
          </div>
          <div
            class="calculator-result-help-date"
          >
            - Data da última atualização: 10/01/2024
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
